packages/ragbits-core/src/ragbits/core/llms/base.py

```diff
@@ -80,12 +82,15 @@ async def generate_raw(
             Raw text response from LLM.
         """
         options = (self.default_options | options) if options else self.default_options
-
+        stream_output = False
+        if hasattr(prompt, "output_type"):
```

Comment:
# 1. 
The `output_type` is just a detail of how the `Prompt` class was implemented. `Prompt` is just one particular way to create prompts in dbally, we also support creating custom prompt classes (for example for prompts backed by an external source - like a database, YAML files, or an external API). That's way the LLM class here declares that it can take any `BasePrompt` object - we should stay true to that declaration. Streaming shouldn't be available exclusively to prompts created in this one particular way.

Moreover, you access `output_type` outside of the regular typing system. If we want prompt classes to be able to contain information whether the prompt should be streamed, we should explicitly declare a method like this in the abstract `BasePrompt` class, so that classes inheriting from that class can provide their implementation.

So, this is how I would implement it (if we really think that streaming should be declared on the level of the prompt - I have my doubts, as described in my other comment):

1. Add a new method to `BasePrompt`:
    ```python
    def is_streamed(self) -> bool:
        False
    ````
2. Add a more specific implementation of the new method to `Prompt`:
    ```python
    def is_streamed(self) -> bool:
        issubclass(self.output_type, tps.AsyncGeneratorType)
    ````
3. Replace like 93 here with `stream=prompt.is_streamed()` and remove lines 85-87

This way any prompt implementation (i.e., any class inheriting with `BasePrompt`) will be able to declare whether the prompt should be streamed, and code specific to how the `Prompt` class is implemented will stay inside `Prompt`.

Reply:
resolved by implementing separate method for streaming